src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Unix.cs

```diff
@@ -77,6 +71,62 @@ private bool InitCultureDataCore()
             return true;
         }
 
+        /// <summary>
+        /// Implementation of culture name validation.
+        /// </summary>
+        /// <remarks>
+        /// This is a fast approximate implementation based on BCP47 spec. It covers only parts of
+        /// the spec; such that, when it returns false, the input is definitely in incorrect format.
+        /// However, it returns true for some characters, which are not allowed by the spec, It also
+        /// returns true for some inputs where spec specifies the lengths of components, but we are
+        /// not validating components individually to limit its complexity at O(n) for now.
+        ///
+        /// Rules of implementation:
+        /// * Allow only letters, digits, - and '_' or \0 (NULL is for backward compatibility).
+        /// * Allow input length of zero (for invariant culture) or otherwise greater than 1 and less than or equal 85.
+        /// * Disallow input that starts or ends with '-' or '_'
+        /// * Disallow input that has any combination of consecutive '-' or '_'.
+        /// * Disallow input that has multiple '_'.
+        /// </remarks>
+        private static bool IsValidCultureName(string subject, out int firstIndexOfUnderscore)
+        {
+            firstIndexOfUnderscore = -1;
+
+            if (subject.Length == 0) return true; // Invariant Culture
+            if (subject.Length == 1 || subject.Length > 85) return false;
+
+            bool seenSeparator = false;
+            bool seenUnderscore = false;
+            for (int i = 0; i < subject.Length; ++i)
+            {
+                char c = subject[i];
+
+                if (c == '_' || c == '-')
+                {
+                    if (i == 0 || i == subject.Length - 1) return false;
+                    if (seenSeparator) return false;
+                    if (c == '_')
+                    {
+                        if (seenUnderscore) return false; // only one _ is allowed
+                        seenUnderscore = true;
+                        firstIndexOfUnderscore = i;
+                    }
+                    seenSeparator = true;
+                }
+                else if (c == '\0' || (c >= '0' && c <= '9') || (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z'))
```

Comment:
maybe can optimize the checks something like 

(uint)((uint)c - (uint) '0') <= 9

and apply same idea for other range checks here.